source/lac/petsc_matrix_base.cc

```diff
@@ -89,14 +89,23 @@ namespace PETScWrappers
     AssertThrow(ierr == 0, ExcPETScError(ierr));
   }
 
+  void
+  MatrixBase::assign_petsc_matrix(Mat A)
+  {
+    AssertThrow(last_action == ::dealii::VectorOperation::unknown,
+                ExcMessage("Cannot assign a new Mat"));
```

Comment:
Out of curiosity, why not?
```suggestion
    AssertThrow(last_action == ::dealii::VectorOperation::unknown,
                ExcMessage("Cannot assign a new Mat."));
```

Reply:
Well, it is a safety check that no operations are in-fly for the previously held matrix.